packages/core/src/common/contribution-filter/contribution-filter-registry.ts

```diff
@@ -0,0 +1,78 @@
+/********************************************************************************
+ * Copyright (C) 2021 STMicroelectronics and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable, multiInject, optional } from 'inversify';
+import { ContributionFilterRegistry, ContributionType, FilterContribution } from './contribution-filter';
+import { applyFilters, Filter } from './filter';
+
+/**
+ * Registry of contribution filters.
+ *
+ * Implement/bind to the `FilterContribution` interface/symbol to register your contribution filters.
+ */
+@injectable()
+export class ContributionFilterRegistryImpl implements ContributionFilterRegistry {
+
+    protected initialized = false;
+    protected genericFilters: Filter<Object>[] = [];
+    protected typeToFilters = new Map<ContributionType, Filter<Object>[]>();
+
+    constructor(
+        @multiInject(FilterContribution) @optional() contributions: FilterContribution[] = []
+    ) {
+        for (const contribution of contributions) {
+            contribution.registerContributionFilters(this);
+        }
+        this.initialized = true;
+    }
+
+    addFilters(types: '*' | ContributionType[], filters: Filter<Object>[]): void {
+        if (this.initialized) {
```

Comment:
I'm not sure we do this check for any other implementations of the registry/contribution callback pattern. Also: I'm not seeing an `this.initialized = true` here ;-)

Reply:
```ts
    constructor(
        @multiInject(FilterContribution) @optional() contributions: FilterContribution[] = []
    ) {
        for (const contribution of contributions) {
            contribution.registerContributionFilters(this);
        }
        this.initialized = true;
    }
```

Reply:
I understand that, but we don't seem to take this precaution in the other instances of this pattern, so why start now and why stop here?

Reply:
Some registries might be designed to be mutated at any time, this one cannot (more precisely it wouldn't have an effect and lead to ignored filters which could be confusing). I'm all for failing as early as possible, so if other places benefit from such checks then yea we should do this there too. But this work is out of scope of this PR I think.

Reply:
At the end of the day it depends on what's the purpose of the component right, in this case it would cause issues to mutate later hence why I locked it with the `initialized` flag.

Reply:
Fair enough.